### PR TITLE
feature(synapse-utilities-common): add one of annotation

### PR DESCRIPTION
**Pull request by brenoreis**
This PR creates an annotation for validation of a set of fields where only one can be not blank.

### CODE REVIEW

File: utility/synapse-utilities-common/src/test/java/io/americanexpress/synapse/utilities/common/validator/OneOfValidatorTest.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+/**
+ * {@code OneOfValidatorTest} unit test class for {@link OneOfValidator}.
+ *
+ * @author Breno Pinto
+ */
+@ExtendWith(MockitoExtension .class)
+class OneOfValidatorTest {
+
+    @Mock
+    private ConstraintValidatorContext constraintValidatorContext;
```
Comment (guttff) [inline utility/synapse-utilities-common/src/test/java/io/americanexpress/synapse/utilities/common/validator/OneOfValidatorTest.java:41]: xtra line

File: utility/synapse-utilities-common/src/main/java/io/americanexpress/synapse/utilities/common/validator/OneOfValidator.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+import javax.validation.ConstraintValidatorContext;
+
+/**
+ * {@code OneOfValidator} validates that only one of the fields is provided.
+ *
+ * @author Breno Pinto
+ */
+public class OneOfValidator implements ConstraintValidator<OneOf, Object> {
+
+    private String[] fieldNames;
```
Comment (guttff) [inline utility/synapse-utilities-common/src/main/java/io/americanexpress/synapse/utilities/common/validator/OneOfValidator.java:34]: javadoc

File: utility/synapse-utilities-common/src/main/java/io/americanexpress/synapse/utilities/common/validator/OneOfValidator.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+     */
+    @Override
+    public boolean isValid(Object object, ConstraintValidatorContext context) {
+
+        if (object == null || ArrayUtils.isEmpty(fieldNames)) {
+            return true;
+        }
+        try {
+            var count = 0;
+            for (String fieldName:fieldNames){
```
Comment (guttff) [inline utility/synapse-utilities-common/src/main/java/io/americanexpress/synapse/utilities/common/validator/OneOfValidator.java]: add the condition to return if found

File: utility/synapse-utilities-common/pom.xml [hunk 4]
```diff
@@ -63,6 +63,10 @@
             <groupId>org.apache.commons</groupId>
             <artifactId>commons-text</artifactId>
         </dependency>
+        <dependency>
+            <groupId>commons-beanutils</groupId>
+            <artifactId>commons-beanutils</artifactId>
+        </dependency>
```
Comment (guttff) [inline utility/synapse-utilities-common/pom.xml]: alphabetize artifact, groupid

File: pom.xml [hunk 5]
```diff
@@ -724,6 +724,11 @@
                 <artifactId>commons-io</artifactId>
                 <version>2.11.0</version>
             </dependency>
+            <dependency>
+                <groupId>commons-beanutils</groupId>
+                <artifactId>commons-beanutils</artifactId>
+                <version>1.9.4</version>
```
Comment (guttff) [inline pom.xml]: move the version to the parent pom

File: pom.xml [hunk 6]
```diff
@@ -724,6 +724,11 @@
                 <artifactId>commons-io</artifactId>
                 <version>2.11.0</version>
```
Comment (guttff) [inline pom.xml]: move the version to the parent pom

File: utility/synapse-utilities-common/src/test/java/io/americanexpress/synapse/utilities/common/validator/OneOfValidatorTest.java (hunk 1 above)
Comment (brenoreis) [reply utility/synapse-utilities-common/src/test/java/io/americanexpress/synapse/utilities/common/validator/OneOfValidatorTest.java:41]: done

File: utility/synapse-utilities-common/src/main/java/io/americanexpress/synapse/utilities/common/validator/OneOfValidator.java (hunk 2 above)
Comment (brenoreis) [reply utility/synapse-utilities-common/src/main/java/io/americanexpress/synapse/utilities/common/validator/OneOfValidator.java:34]: done

File: utility/synapse-utilities-common/src/main/java/io/americanexpress/synapse/utilities/common/validator/OneOfValidator.java (hunk 3 above)
Comment (brenoreis) [reply utility/synapse-utilities-common/src/main/java/io/americanexpress/synapse/utilities/common/validator/OneOfValidator.java]: done

File: utility/synapse-utilities-common/pom.xml (hunk 4 above)
Comment (brenoreis) [reply utility/synapse-utilities-common/pom.xml]: done

File: pom.xml (hunk 5 above)
Comment (brenoreis) [reply pom.xml]: done

File: pom.xml (hunk 6 above)
Comment (brenoreis) [reply pom.xml]: done

**Review by guttff — APPROVED** [review]

**Pull request merged**